compiler/rustc_ast/src/ast.rs

```diff
@@ -779,8 +779,8 @@ pub enum PatKind {
     Ident(BindingAnnotation, Ident, Option<P<Pat>>),
 
     /// A struct or struct variant pattern (e.g., `Variant {x, y, ..}`).
-    /// The `bool` is `true` in the presence of a `..`.
-    Struct(Option<P<QSelf>>, Path, ThinVec<PatField>, /* recovered */ bool),
+    /// The `bool` is `true` in the presence of a `..` (or when recovered).
+    Struct(Option<P<QSelf>>, Path, ThinVec<PatField>, bool),
```

Comment:
Please either convert this into:

```
Struct {
    qself: Option<P<QSelf>>,
    path: Path,
    fields: ThinVec<PatField>,
    // Whether `..` is meant to capture the rest of the tuple fields
    rest: bool,
}
```

Or alternatively, change the bool into an enum like `StructRest` is used in `StructExpr`:

```
enum ??idk_what_to_call_this?? {
  Rest,
  None,
}
```

Do you think you could do either of these? I have no preference which, but I think we could do better 😸

Reply:
I think the latter is preferable, as it also clarifies it in the return type in the parser.

Reply:
Should I make the same change in HIR? Or can I leave that as a potential followup commit?